src/jemalloc.c

```diff
@@ -1855,7 +1860,8 @@ malloc_init_hard_a0_locked(void) {
 	if (config_prof) {
 		prof_boot0();
 	}
-	malloc_conf_init(&sc_data, bin_shard_sizes);
+	char readlink_buf[PATH_MAX + 1];
+	malloc_conf_init(&sc_data, bin_shard_sizes, readlink_buf);
```

Comment:
Question: what if the setting is a string longer than PATH_MAX? If so, would the overflow be a concern? Can we add a test to confirm the behavior under this scenario is as expected?

Reply:
We explicitly add the terminator here: https://github.com/jemalloc/jemalloc/pull/2601/files#diff-a4cb09e38cfec8141b07c291f731a8e01a17412568a852884fd921e8e521766bR1015, which means we will truncate the buffer in case of > PATH_MAX which is by design.

re: tests -- I'll leave it to Sherry to decide if a test is worth it.